### PR TITLE
Add LoadResourceNames function to parse all resource names in a workspace. Closes #813

### LINKED ISSUE #813: Add LoadResourceNames function to parse all resource names in a workspace

### PR DESCRIPTION
**Pull request by kaidaguerre**
Added unit tests.

### CODE REVIEW

File: steampipeconfig/load_mod_test.go [hunk 1]
```diff
@@ -408,3 +409,67 @@ func executeLoadTest(t *testing.T, name string, test loadModTest, wd string) {
...
 }
+
+type loadResourceNamesTest struct {
+	source   string
+	expected interface{}
+}
+
+var testCasesLoadResourceNames map[string]loadResourceNamesTest
+
+func init() {
```
Comment (kaidaguerre) [inline steampipeconfig/load_mod_test.go:420]: you can actually just define the tests cases in the global var, the same as the workspace test I linked - the mod load test is a special case

File: steampipeconfig/load_mod_test.go [hunk 2]
```diff
@@ -408,3 +409,67 @@ func executeLoadTest(t *testing.T, name string, test loadModTest, wd string) {
...
+		}
+
+		if test.expected == "ERROR" {
+			t.Errorf("Test: '%s'' FAILED - expected error", name)
+			continue
+		}
+
+		// to compare the benchmarks
+		benchmark_expected := test.expected.(*modconfig.WorkspaceResources).Benchmark
+		flag_b := reflect.DeepEqual(names.Benchmark, benchmark_expected)
```
Comment (kaidaguerre) [inline steampipeconfig/load_mod_test.go:453]: no need for flag, you can just uses 
```
if reflect.DeepEqual(names.Benchmark, benchmark_expected){
}
```

File: steampipeconfig/load_mod_test.go [hunk 3]
```diff
@@ -408,3 +409,67 @@ func executeLoadTest(t *testing.T, name string, test loadModTest, wd string) {
...
+		if flag_b != true {
+			t.Log(`"expected" is not equal to "output"`)
+			t.Errorf("FAILED \nexpected: %#v\noutput: %#v", benchmark_expected, names.Benchmark)
+		}
+
+		// to compare the controls
+		control_expected := test.expected.(*modconfig.WorkspaceResources).Control
+		flag_c := reflect.DeepEqual(names.Control, control_expected)
```
Comment (kaidaguerre) [inline steampipeconfig/load_mod_test.go:461]: again, no need for flag

**Review by kaidaguerre — COMMENTED** [review]
A few small changes

File: steampipeconfig/load_mod_test.go (hunk 3 above)
Comment (pskrbasu) [reply steampipeconfig/load_mod_test.go]: Made the change in e16f0936849f0f5531d2ad7b686d66daf2026cd3

File: steampipeconfig/load_mod_test.go (hunk 2 above)
Comment (pskrbasu) [reply steampipeconfig/load_mod_test.go]: Made the change in e16f0936849f0f5531d2ad7b686d66daf2026cd3

File: steampipeconfig/load_mod_test.go (hunk 1 above)
Comment (pskrbasu) [reply steampipeconfig/load_mod_test.go]: Made the change in e16f0936849f0f5531d2ad7b686d66daf2026cd3

**Pull request merged**